modules/ign.js

```diff
@@ -0,0 +1,220 @@
+const { GoogleSpreadsheet } = require('google-spreadsheet');
+const sf = require("../config/snowflakes.json");
+
+/** @type {discord.Collection<string, ign>} */
+let IGNs = new u.Collection();
+let aliases = new u.Collection();
+
+const findSystem = (system) => IGNs.get(system?.toLowerCase()) || IGNs.get(aliases.get(system?.toLowerCase()));
+const categories = [
+  "Game Platforms",
+  "Streaming",
+  "Social",
+  "Personal"
+];
+
+/**
+ * @typedef ign
+ * @prop {string} system
+ * @prop {string[]} aliases
+ * @prop {string} name
+ * @prop {string} category
+ * @prop {string} link
+ */
+
+/**
+ * Creates and formats the embed for the IGN system.
+ * @param {discord.GuildMember} user The member that we're displaying the command for.
+ * @param {string[]} systems A mapping of system: username/ign
+ * @returns {discord.MessageEmbed} the embed to send
+ */
+function createIgnEmbed(user, igns, systems) {
+  const embed = u.embed().setAuthor({ name: user.displayName, iconURL: user.displayAvatarURL });
+  if (systems.length == 0) return null;
+  if (systems.length > 1) { embed.setTitle('IGNs for ' + u.escapeText(user.displayName)); }
+  const hasLink = /(http(s?):\/\/)?(\w+\.)+\w+\//ig;
+  const mapped = systems.map(s => findSystem(s)).filter(s => s != null);
+  if (mapped.length == 0) return null;
+  for (const category of categories) {
+    const sys = mapped.filter(s => s.category == category)
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .map((system, i) => {
+        if (i > 24) return null;
+        let name = igns.find(ign => ign.system == system.system)?.ign;
+        if (!name) return null;
+        if (name.length > 100) name = name.substr(0, 100) + " ...";
+        if (system.link && !hasLink.test(name)) name = `[${name}](${system.link.replace(/{ign}/ig, encodeURIComponent(name))})`;
+        embed.addField(`${system.name}`, `${name}`, true);
+      });
+    sys;
+  }
+  return embed.fields.length > 0 ? embed : null;
+}
+/** @param {discord.CommandInteraction} interaction */
+async function slashIgnView(interaction) {
+  const user = interaction.options.getMember("target", false) || interaction.member;
+  let system = interaction.options.getString("system", false);
+  if (findSystem(system)) system = findSystem(system).system;
+  const igns = await Module.db.ign.find(user.id, system);
+
+  const embed = createIgnEmbed(user, Array.isArray(igns) ? igns : [igns].filter(i => i != null), system ? [system] : igns.map(i => i.system));
+
+  if (embed) {
+    interaction.reply({ embeds: [embed] });
+  } else {
+    interaction.reply({ content: `It looks like ${user.displayName} hasn't saved this information with \`/ign add\` yet.`, ephemeral: true });
+  }
+
+}
+/** @param {discord.CommandInteraction} interaction */
+async function slashIgnSet(interaction) {
+  let system = interaction.options.getString("system").toLowerCase();
+  let ign = interaction.options.getString("ign").toLowerCase();
```

Comment:
```suggestion
  let ign = interaction.options.getString("ign");
```
Let's not force IGNs into lowercase.